torch/csrc/distributed/c10d/logger.cpp

```diff
@@ -316,7 +316,9 @@ void Logger::set_runtime_stats_and_log() {
     );
     return;
   }
-  if (!reducer_->params_[0].is_cuda() && !reducer_->params_[0].is_cpu()) {
+
+  if (!reducer_->timer_ &&
```

Comment:
Could you add the reasoning for the change in the PR description? This change means that we will never hit the ASSERT on line 329

Reply:
@H-Huang add the reasoning for the change. Without this change also, the ASSERT on line 329, will never be reported, because of this check for other backends if (!reducer_->params_[0].is_cuda() && !reducer_->params_[0].is_cpu()) {...} it reports a warn msg and returns. (i understood, this case  is required for lazy backend). to make generic - add  a check to do only if timer is registered.

Reply:
@jeejakp12, I do not follow. Shouldn't this check for timer be a separate check and include a separate warning message then? How come it is added to `if (!reducer_->params_[0].is_cuda() && !reducer_->params_[0].is_cpu()) {...}`

Reply:
@H-Huang  if the check for timer has to separate, then it is expected that all backends has to register for this timer which is not valid case there are backend which does not register for this timer.  only for CPU and CUDA it is has mandatory timer registration. Now the HPU backend also have register for the timer, in which case it return with warning because the check only include cpu and cuda, i can add HPU also there, but it is not generic. That is why if timer is not registered, then check if for backend is not CPU and CUDA, then return